models/issue_dependency.go

```diff
@@ -0,0 +1,159 @@
+	sess := x.NewSession()
+
+	// Check if it aleready exists
+	exists, err = issueDepExists(x, issue.ID, dep.ID)
+	if err != nil {
+		return exists, err
+	}
+
+	// If it not exists, create it, otherwise show an error message
+	if !exists {
+		newIssueDependency := &IssueDependency{
+			UserID:       user.ID,
+			IssueID:      issue.ID,
+			DependencyID: dep.ID,
+		}
+
+		if _, err := x.Insert(newIssueDependency); err != nil {
+			return exists, err
+		}
+
+		// Add comment referencing the new dependency
+		_, err = createIssueDependencyComment(sess, user, issue, dep, true)
+
+		if err != nil {
+			return exists, err
+		}
+
+		// Create a new comment for the dependent issue
+		_, err = createIssueDependencyComment(sess, user, dep, issue, true)
+
+		if err != nil {
+			return exists, err
+		}
+	}
+	return exists, nil
+}
+
+// RemoveIssueDependency removes a dependency from an issue
+func RemoveIssueDependency(user *User, issue *Issue, dep *Issue, depType DependencyType) (err error) {
+	sess := x.NewSession()
+
+	// Check if it exists
+	exists, err := issueDepExists(x, issue.ID, dep.ID)
+	if err != nil {
+		return err
+	}
+
+	// If it exists, remove it, otherwise show an error message
+	if exists {
+
+		var issueDepToDelete IssueDependency
+
+		switch depType {
+		case DependencyTypeBlockedBy:
+			issueDepToDelete = IssueDependency{IssueID: issue.ID, DependencyID: dep.ID}
+		case DependencyTypeBlocking:
+			issueDepToDelete = IssueDependency{IssueID: dep.ID, DependencyID: issue.ID}
+		default:
+			return
+		}
+
+		_, err := x.Delete(&issueDepToDelete)
+		if err != nil {
+			return err
+		}
+
+		// Add comment referencing the removed dependency
+		_, err = createIssueDependencyComment(sess, user, issue, dep, false)
+
+		if err != nil {
+			return err
+		}
+
+		// Create a new comment for the dependent issue
+		_, err = createIssueDependencyComment(sess, user, dep, issue, false)
+
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Check if the dependency already exists
+func issueDepExists(e Engine, issueID int64, depID int64) (exists bool, err error) {
+
+	deps := new(IssueDependency)
+	exists, err = e.Where("(issue_id = ? AND dependency_id = ?) OR (issue_id = ? AND dependency_id = ?)", issueID, depID, depID, issueID).Get(deps)
+
+	if err != nil {
+		return exists, err
+	}
+
+	return exists, nil
+}
+
+// IssueDependencyIssue custom type for mysql join
+type IssueDependencyIssue struct {
+	IssueDependency `xorm:"extends"`
+	Issue           `xorm:"extends"`
+}
+
+// TableName returns table name for mysql join
+func (IssueDependencyIssue) TableName() string {
+	return "issue_dependency"
+}
+
+// IssueNoDependenciesLeft checks if issue can be closed
+func IssueNoDependenciesLeft(issue *Issue) bool {
+
+	total, err := x.
+		Join("INNER", "issue", "issue.id = issue_dependency.issue_id").
+		Where("issue_dependency.issue_id = ?", issue.ID).
+		And("issue.is_closed = ?", "0").
+		Count(&IssueDependencyIssue{})
```

Comment:
I think you can use `Exist(&IssueDependencyIssue{})` instead and return result if err is empty.

Reply:
Wouldn't it work with `&IssueDependency{}` (I am not sure if it works in xorm this way)?

Reply:
Ok, how does this work? I couldn't find documentation about this (in xorms documentation).

Also this seems to me more like a hack than a solution.

Reply:
It's on github xorm main page. I don't see any hack here,

Reply:
Such construction is used in many places in code. It is used to know what table name to use in `from` part of sql

Reply:
Also if you set fields, it will use it in `where`.

Reply:
Changed it as you proposed.